src/main/java/ie/gmit/softwareeng/labexam/Playlist.java

```diff
@@ -1,71 +1,98 @@
 package ie.gmit.softwareeng.labexam;
 
+import java.util.ArrayList;
+import java.util.Collections;
+
 public class Playlist {
+    private String playlistName;
+    ArrayList<Song> songs = new ArrayList<Song>();
+    Song currentSong;
```

Comment:
This works but is a bit complicated. You could have tracked the current song index instead and initialised it to 0. This would make addSong a bit simpler